packages/recs/src/Component.ts

```diff
@@ -374,6 +374,7 @@ export function overridableComponent<S extends Schema, M extends Metadata, T = u
       if (prop === "removeOverride") return removeOverride;
       if (prop === "values") return valuesProxy;
       if (prop === "update$") return update$;
+      if (prop === "entities") return () => [...overriddenEntityValues.keys(), ...target.entities()];
```

Comment:
maybe we could use `concatIterators(overriddenEntityValues.keys(), target.entities())` here (from `@latticexyz/utils`) to avoid spreading the values intro an array?

Reply:
This gives type errors elsewhere. We're already using the spread operator above in a similar way. I think this is fine?

Is there something about spread you're worried about?

Reply:
I was worried about the performance overhead (O(n) for spreading vs O(1) for creating a new iterator), but you're raising a valid point - above we have to spread into a set to avoid duplicate entities (`overriddenEntityValues.keys()` might contain keys also included in `target.entities()`), I think we'll have to do the same here

```suggestion
      if (prop === "entities") return () => new Set([...overriddenEntityValues.keys(), ...target.entities()]).values();
```